microgrid_optimizer.py

```diff
@@ -922,33 +931,47 @@ def parse_results(self):
         if not len(self.output_system_grid):
             self.run_sims()
 
-        # Results columns
-        metrics = ['pv_capacity', 'battery_capacity', 'battery_power',
-                   'generator_power', 'fuel_tank_size_gal',
-                   'capital_cost_usd', 'pv_capital', 'battery_capital',
-                   'generator_capital', 'fuel_tank_capital', 'pv_o&m',
-                   'battery_o&m', 'generator_o&m', 'pv_area_ft2',
-                   'annual_benefits_usd', 'demand_benefits_usd',
-                   'simple_payback_yr', 'pv_percent mean',
-                   'batt_percent mean', 'gen_percent mean',
-                   'generator_power_kW mean', 'generator_power_kW std',
-                   'generator_power_kW most-conservative',
-                   'fuel_used_gal mean', 'fuel_used_gal std',
-                   'fuel_used_gal most-conservative']
-
-        # Add columns for displaying information about smaller generator sizes
-        for perc in self.gen_power_percent:
-            metrics += ['{}%_smaller_gen_size'.format(perc),
-                        '{}%_smaller_gen_typical_fuel_gal'.format(perc),
-                        '{}%_smaller_gen_conservative_fuel_gal'.format(perc),
-                        '{}%_smaller_gen_cost'.format(perc),
-                        '{}%_smaller_gen_scenarios_not_met'.format(perc),
-                        '{}%_smaller_gen_hours_not_met_average'.format(perc),
-                        '{}%_smaller_gen_hours_not_met_max'.format(perc),
-                        '{}%_smaller_gen_kWh_not_met_average'.format(perc),
-                        '{}%_smaller_gen_kWh_not_met_max'.format(perc),
-                        '{}%_smaller_gen_max_%_kW_not_met_average'.format(perc),
-                        '{}%_smaller_gen_max_%_kW_not_met_max'.format(perc)]
+        if not self.existing_generator:
```

Comment:
Rather than duplicating most of the list here, you could just append the relevant elements that are unique to each conditional.